openedx/openedx-platform
# feat: ESG BFF /initialize

## Description

Add the first call (`initialize`) for the Enhanced Staff Grader (ESG) backend-for-frontend (BFF).

JIRA: [https://openedx.atlassian.net/browse/AU-305](https://openedx.atlassian.net/browse/AU-305)
Note: Relies on `list_staff_workflows` XBlock handler to be completed from [AU-53](https://openedx.atlassian.net/browse/AU-53)

## Testing instructions

1. Get or mock required XBlock endpoint in ORA (`list_staff_workflow`).
2. Using Postman or another API testing tool, authenticate with LMS as a staff user (you may find it useful to follow [these instructions on how to authenticate with Postman](https://openedx.atlassian.net/wiki/spaces/ENG/pages/3145892500/How+to+authenticate+and+query+edX+APIs+with+Postman) to get set up).
3. Using your API testing tool, hit the endpoint below as an authenticated staff user `GET http://{{lms_url}}/api/ora_staff_grader/initialize?course_id={{course_id_encoded}}&ora_location={{block_id_encoded}}`
4. Verify that the returned data is accurate and matches the [data contract](https://openedx.atlassian.net/wiki/spaces/PT/pages/3154542730/Enhanced+Staff+Grader+Data+Flow+Design#Initialize).

FYI: @edx/masters-devs-gta
